### PR TITLE
Add another patch for numpy 2.0 (main branch)

### PR DESCRIPTION
**Pull request by bkpoon**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

This is the same as #211 but for the `main` branch.

### CODE REVIEW

**Comment by bkpoon** [comment]
@conda-forge-admin please rerender

**Comment by bkpoon** [comment]
@conda-forge-admin please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -24,12 +24,14 @@ source:
...
     - patches/0001-Add-default-value-for-cxx-and-cxxflags-options-for-t.patch
     # backport https://github.com/boostorg/charconv/pull/183
     - patches/0002-Merge-pull-request-183-from-boostorg-ppc64le.patch
     # backport https://github.com/boostorg/python/pull/432
     - patches/0003-Support-numpy-2.0.0b1.patch
+    # backport https://github.com/boostorg/python/pull/443
+    - patches/0004-Support-numpy-2.0.patch
 
 build:
   number: 3
```
Comment (h-vetinari) [inline recipe/meta.yaml:37]:
```suggestion
  number: 4
```

**Comment by bkpoon** [comment]
Checking in to see if anything else needs to be done. Thanks!

**Review by h-vetinari — APPROVED** [review]

**Comment by bkpoon** [comment]
Thanks! Feel free to ping me if there are any issues.

**Pull request merged**